framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -770,35 +805,27 @@ public AnnotationFormatter getAnnotationFormatter() {
     // Factories for annotated types that account for implicit qualifiers
     // **********************************************************************
 
-    /** Should results be cached? Disable for better debugging. */
-    protected static boolean SHOULD_CACHE = true;
-    public boolean shouldCache = SHOULD_CACHE;
-
-    /** Should the cached result be used, or should it be freshly computed? */
-    protected static boolean SHOULD_READ_CACHE = true;
-    public boolean shouldReadCache = SHOULD_READ_CACHE;
-
-    /** Size of LRU cache. */
-    private final static int CACHE_SIZE = 300;
-
-    /** Mapping from a Tree to its annotated type; implicits have been applied. */
-    private final Map<Tree, AnnotatedTypeMirror> treeCache = CollectionUtils.createLRUCache(CACHE_SIZE);
-
-    /** Mapping from a Tree to its annotated type; before implicits are applied,
-     * just what the programmer wrote. */
-    protected final Map<Tree, AnnotatedTypeMirror> fromTreeCache = CollectionUtils.createLRUCache(CACHE_SIZE);
-
-    /** Mapping from an Element to its annotated type; before implicits are applied,
-     * just what the programmer wrote. */
-    private final Map<Element, AnnotatedTypeMirror> elementCache = CollectionUtils.createLRUCache(CACHE_SIZE);
-
-    /** Mapping from an Element to the source Tree of the declaration. */
-    private final Map<Element, Tree> elementToTreeCache  = CollectionUtils.createLRUCache(CACHE_SIZE);
-
     /** Mapping from a Tree to its TreePath **/
     private final TreePathCacher treePathCache = new TreePathCacher();
 
     /**
+     * Returns the int supplied to the checker via atfCacheSize or
+     * the default cache size.
+     * @return cache size passed as argument to checker or DEFAULT_CACHE_SIZE
+     */
+    private int getCacheSize() {
+        String option = checker.getOption("atfCacheSize");
+        if(option == null)
```

Comment:
Recommend "if (option != null) { ... }" so that "return DEFAULT_CACHE_SIZE" can be written once.